controller/src/util/config_action.rs

```diff
@@ -0,0 +1,429 @@
+                Some(prev_config_state) => match &config.spec.broker_spec {
+                    None => {
+                        config_map.lock().await.insert(
+                            config.metadata.name.as_ref().unwrap().to_string(),
+                            prev_config_state,
+                        );
+                    }
+                    Some(broker) => {
+                        if !should_recreate_config(&config, &prev_config_state) {
+                            trace!("handle_config - should redeploy broker");
+                            config_map.lock().await.insert(
+                                config.metadata.name.as_ref().unwrap().to_string(),
+                                new_config_state,
+                            );
+                            match broker {
+                                BrokerSpec::BrokerPodSpec(p) => {
+                                    handle_broker_updates_for_configuration_pod(
+                                        kube_interface,
+                                        &config,
+                                        p,
+                                    )
+                                    .await?;
+                                }
+                                BrokerSpec::BrokerJobSpec(j) => {
+                                    handle_broker_updates_for_configuration_job(
+                                        kube_interface,
+                                        *config.metadata.generation.as_ref().unwrap(),
+                                        &config,
+                                        j,
+                                    )
+                                    .await?;
+                                }
+                            }
+                        }
+                    }
+                },
+                None => {
+                    config_map.lock().await.insert(
+                        config.metadata.name.as_ref().unwrap().to_string(),
+                        new_config_state,
+                    );
+                }
+            }
+        }
+        Event::Deleted(config) => {
+            info!(
+                "handle_config - deleted Configuration {:?}",
+                config.metadata.name,
+            );
+            // Remove Configuration from map
+            config_map
+                .lock()
+                .await
+                .remove(config.metadata.name.as_ref().unwrap());
+        }
+        Event::Restarted(_configs) => {
+            if *first_event {
+                info!("handle_config - watcher started");
+            } else {
+                return Err(anyhow::anyhow!(
+                    "Configuration watcher restarted - throwing error to restart agent"
+                ));
+            }
+        }
+    }
+    *first_event = false;
+    Ok(())
+}
+
+/// This function is called when the Configuration::BrokerSpec was modified where
```

Comment:
nit: not sure i understand this sentence ... should "where BrokerSpec::BrokerPodSpec" be removed?

Reply:
its trying to say that its called when the BrokerSpec is type BrokerPodSpec. i can reword